snap/info.go

```diff
@@ -943,6 +943,7 @@ type AppInfo struct {
 	RestartDelay    timeout.Timeout
 	Completer       string
 	RefreshMode     string
+	RefreshAllowed  bool
```

Comment:
probably needs input from @pedronis as discussed elsewhere maybe extending RefreshMode (endure?) would be a good idea, but then it probably needs some checks to validate refresh-mode for apps and non-apps

Reply:
my preference is to support "refresh-mode: ignore-running" for apps. This will make yaml validation fail on old snapd (as they allow refresh-mode only on services) so we also need to introduce an assumes: app-refresh-mode

Reply:
Thanks, I'll follow up with that.